cogs5e/models/automation/effects/target.py

```diff
@@ -27,8 +30,23 @@ def run(self, autoctx):
         previous_target = autoctx.target
         result_pairs = []
 
+        targets = autoctx.targets
+        if isinstance(self.sort_by, str):
+            if self.sort_by.endswith('desc'):
+                descending = True
+                sort_last = float('-inf')
+            else:
+                descending = False
+                sort_last = float('inf')
+
+            if self.sort_by in ('hp_asc', 'hp_desc'):
+                targets = sorted(
+                    autoctx.targets, reverse=descending,
+                    key=lambda t: (t.hp or sort_last) if isinstance(t, StatBlock) else sort_last
+                )
```

Comment:
```suggestion
        if self.sort_by == 'hp_asc':
            targets = sorted(
                autoctx.targets, 
                key=lambda t: (t.hp or float('inf')) if isinstance(t, StatBlock) else float('inf')
            )
        elif self.sort_by == 'hp_desc':
            targets = sorted(
                autoctx.targets,
                key=lambda t: (t.hp or float('-inf')) if isinstance(t, StatBlock) else float('-inf'),
                reverse=True
            )
```
Clever, but it's actually shorter and more readable to just do a simple if-switch 😛 (and maybe more so if you pull out the `key` into a function in utils, like `target_hp_or_default(target, default)` - this would also let you get rid of the nasty edge case where a `None` hp combatant could be sorted in front of a `0` hp combatant as both `None` and `0` are falsy)

Reply:
Fair. I have made these changes, including the function, but can't test them until I get back home. Good catch with the `0`